modules/tensor_mechanics/include/interfacekernels/ADCZMInterfaceKernel.h

```diff
@@ -0,0 +1,35 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ADInterfaceKernel.h"
+
+/// AD equivalent of CZMInterfaceKernelBase
+class ADCZMInterfaceKernel : public ADInterfaceKernel
```

Comment:
why not call it ```ADCZMInterfaceKernelBase```?

Reply:
Because the AD version, `ADCZMInterfaceKernel`, works for both small strain and large deformation, and so it is being registered.